File: scripts/enhance_batch_300_349.py
#!/usr/bin/env python3
import json
import os
from datetime import datetime
from pathlib import Path

# Paths
SCRIPT_DIR = Path(__file__).parent
CONTENT_DIR = SCRIPT_DIR.parent / 'content' / 'products'
TEMPLATES_FILE = SCRIPT_DIR / 'templates-to-enhance.json'

# Load templates to enhance
with open(TEMPLATES_FILE, 'r') as f:
    all_templates = json.load(f)

# Extract batch 300-349
batch = all_templates[300:350]

print(f"Processing batch 300-349: {len(batch)} templates\n")

enhanced_count = 0
skipped_count = 0
errors = []

def generate_enhanced_content(product):
    """Generate comprehensive enhanced content for a product."""
    name = product.get('name', '')
    description = product.get('description', '')[:600]
    subcategory = product.get('subcategory', '')
    price = product.get('price', 0)

    # Generate content
    enhanced = {
        'overview': generate_overview(name, description, subcategory, price),
        'featureCategories': generate_feature_categories(name, subcategory),
        'problemsSolved': generate_problems(subcategory),
        'solutionsBenefits': generate_solutions(subcategory),
        'idealFor': generate_ideal_for(subcategory),
        'notRecommendedFor': [
            'Users who prefer physical/paper-based systems',
            'People uncomfortable with digital tools',
            'Those looking for a mobile-first solution (Notion mobile has limitations)',
            'Users who need offline access as primary feature'
        ],
        'complexityLevel': generate_complexity(price),
        'faqs': generate_faqs(name, subcategory)
    }

    return enhanced

def generate_overview(name, description, subcategory, price):
    """Generate 2-3 paragraph overview."""
    clean_desc = description.replace('\n', ' ').strip()

    return [
        f"The {name} is a comprehensive Notion {subcategory} template designed to streamline your workflow and enhance productivity. {clean_desc[:200] if clean_desc else 'This template provides all the essential tools you need to stay organized and achieve your goals with ease.'}",

        f"Built with both beginners and experienced Notion users in mind, this {subcategory} template features an intuitive interface and well-organized structure. The template is fully customizable, allowing you to adapt it to your specific needs while maintaining a professional, polished appearance.",

        f"{'Available as a free resource, this template delivers exceptional value without any financial commitment.' if price == 0 else f'At ${price}, this template offers excellent value for anyone serious about improving their {subcategory} workflow.'} Whether you're managing personal projects or professional work, this template provides the foundation you need to succeed."
    ]

def generate_feature_categories(name, subcategory):
    """Generate 4 feature categories with icons."""
    return [
        {
            'name': 'Core Features',
            'icon': '⚡',
            'features': [
                f'Comprehensive {subcategory} management system',
                'Intuitive database structure with multiple views',
                'Customizable fields and properties',
                'Easy-to-use interface for quick navigation'
            ]
        },
        {
            'name': 'Organization',
            'icon': '📊',
            'features': [
                'Multiple database views (table, board, calendar, gallery)',
                'Advanced filtering and sorting capabilities',
                'Tag and category system for easy classification',
                'Quick search and navigation features'
            ]
        },
        {
            'name': 'Productivity Tools',
            'icon': '🎯',
            'features': [
                'Template buttons for quick entry creation',
                'Automated formulas and calculations',
                'Progress tracking and status indicators',
                'Integration-ready database structure'
            ]
        },
        {
            'name': 'Customization',
            'icon': '🎨',
            'features': [
                'Fully customizable layout and design',
                'Flexible database properties',
                'Aesthetic and functional design elements',
                'Adaptable to your specific workflow needs'
            ]
        }
    ]

def generate_problems(subcategory):
    """Generate problems solved based on subcategory."""
    problem_map = {
        'tracker': [
            'Difficulty keeping track of multiple items across different platforms',
            'Lost or forgotten information',
            'Scattered data without centralized organization',
            'Lack of visibility into patterns and trends'
        ],
        'planner': [
            'Poor time management and scheduling conflicts',
            'Forgotten tasks and missed deadlines',
            'Inability to prioritize effectively',
            'Lack of clear overview of commitments'
        ],
        'dashboard': [
            'Information scattered across multiple tools',
            'No centralized view of important data',
            'Time wasted switching between applications',
            'Difficulty maintaining project oversight'
        ],
        'productivity': [
            'Low productivity and time wastage',
            'Unclear priorities and goals',
            'Difficulty maintaining focus on important work',
            'Lack of systematic progress tracking'
        ],
        'finance': [
            'Poor visibility into spending patterns',
            'Difficulty budgeting and achieving savings goals',
            'Disorganized financial records',
            'Lack of strategic financial planning'
        ],
        'journal': [
            'Inconsistent journaling habits',
            'Difficulty organizing thoughts and reflections',
            'Lost journal entries across multiple platforms',
            'Lack of structure for meaningful reflection'
        ],
        'goals': [
            'Unclear or undefined goals',
            'Difficulty tracking progress toward objectives',
            'Lack of accountability systems',
            'Goals forgotten or abandoned'
        ],
        'crm': [
            'Disorganized contact and customer information',
            'Poor relationship management',
            'Missed follow-ups and opportunities',
            'Lack of visibility into customer interactions'
        ],
        'content': [
            'Inconsistent content creation and publishing',
            'Difficulty brainstorming and organizing ideas',
            'Poor content calendar management',
            'Lack of content strategy and planning'
        ],
        'student': [
            'Overwhelming coursework and assignments',
            'Difficulty tracking multiple class schedules',
            'Disorganized study materials and notes',
            'Poor exam preparation and time management'
        ],
        'design': [
            'Scattered design assets and resources',
            'Difficulty managing design projects',
            'Poor version control and organization',
            'Lack of centralized design system'
        ],
        'project-management': [
            'Projects running over budget or timeline',
            'Poor team coordination and communication',
            'Lack of visibility into project status',
            'Difficulty managing multiple projects simultaneously'
        ],
        'reading': [
            'Books forgotten after completion',
            'Difficulty tracking reading progress',
            'No system for capturing book insights',
            'Lost book recommendations and notes'
        ],
        'second-brain': [
            'Information overload and mental clutter',
            'Forgotten ideas and valuable knowledge',
            'Difficulty connecting related concepts',
            'Inefficient personal knowledge management'
        ],
        'ai': [
            'Inefficient workflows without AI optimization',
            'Lack of structured AI tool integration',
            'Difficulty maximizing AI capabilities',
            'Scattered AI resources and prompts'
        ]
    }

    return problem_map.get(subcategory, [
        f'Disorganized {subcategory} management',
        'Time wasted on manual, repetitive processes',
        'Lack of systematic approach to workflow',
        'Difficulty maintaining consistency and quality'
    ])

def generate_solutions(subcategory):
    """Generate solutions/benefits based on subcategory."""
    solution_map = {
        'tracker': [
            'Complete visibility of all tracked items in one centralized location',
            'Never lose important information again',
            'Identify patterns and trends to make better decisions',
            'Save time with organized, accessible data'
        ],
        'planner': [
            'Clear overview of your schedule and all commitments',
            'Never miss important deadlines or events',
            'Prioritize tasks effectively for maximum impact',
            'Reduce stress with better planning and organization'
        ],
        'dashboard': [
            'All critical information accessible from one central hub',
            'Save hours by eliminating constant tool-switching',
            'Make faster, better-informed decisions',
            'Maintain complete oversight of all your projects'
        ],
        'productivity': [
            'Dramatically boost your productivity and output',
            'Focus on what truly matters with clear priorities',
            'Track progress and celebrate achievements',
            'Build sustainable productive habits'
        ],
        'finance': [
            'Take complete control of your financial situation',
            'Make informed decisions backed by data',
            'Achieve savings goals faster and more reliably',
            'Reduce financial stress and build confidence'
        ],
        'journal': [
            'Develop a consistent, rewarding journaling practice',
            'Organize thoughts and gain clarity',
            'Track personal growth over time',
            'Create a searchable archive of your reflections'
        ],
        'goals': [
            'Set clear, achievable goals with defined milestones',
            'Track progress systematically toward your objectives',
            'Stay motivated with visible achievements',
            'Turn aspirations into accomplished reality'
        ],
        'crm': [
            'Build stronger, more meaningful relationships',
            'Never miss important follow-ups or opportunities',
            'Maintain complete contact and interaction history',
            'Grow your network strategically'
        ],
        'content': [
            'Publish consistently with organized planning',
            'Never run out of content ideas',
            'Streamline content creation workflow',
            'Build an engaged audience with strategic content'
        ],
        'student': [
            'Excel academically with organized study systems',
            'Stay on top of all assignments and deadlines',
            'Reduce academic stress and anxiety',
            'Achieve better grades with less effort'
        ],
        'design': [
            'Access design assets instantly when needed',
            'Maintain consistency across all design work',
            'Collaborate effectively with team members',
            'Build and maintain comprehensive design systems'
        ],
        'project-management': [
            'Deliver projects on time and within budget',
            'Improve team coordination and productivity',
            'Maintain complete project visibility',
            'Successfully manage multiple complex projects'
        ],
        'reading': [
            'Remember and apply book insights long-term',
            'Track reading progress and set goals',
            'Build a personal library of wisdom',
            'Never forget what you read'
        ],
        'second-brain': [
            'Capture and organize all your valuable knowledge',
            'Never forget important ideas or insights',
            'Connect concepts to spark creativity',
            'Build a powerful personal knowledge repository'
        ],
        'ai': [
            'Maximize AI tools for peak productivity',
            'Access organized AI resources instantly',
            'Streamline workflows with AI integration',
            'Stay ahead with cutting-edge AI strategies'
        ]
    }

    return solution_map.get(subcategory, [
        f'Streamline your entire {subcategory} workflow',
        'Increase efficiency and reduce wasted effort',
        'Achieve better results with systematic organization',
        'Gain peace of mind with reliable systems'
    ])

def generate_ideal_for(subcategory):
    """Generate ideal user profiles."""
    ideal_map = {
        'tracker': [
            'Anyone managing multiple items or activities regularly',
            'Detail-oriented individuals who value organization',
            'People wanting to identify patterns in their data',
            'Users seeking centralized tracking solutions'
        ],
        'planner': [
            'Busy professionals juggling multiple commitments',
            'Students balancing coursework and activities',
            'Anyone wanting to improve time management',
            'People who prefer digital planning tools'
        ],
        'dashboard': [
            'Professionals needing quick access to key information',
            'Project managers coordinating multiple initiatives',
            'Entrepreneurs managing various business aspects',
            'Anyone valuing centralized workspace organization'
        ],
        'productivity': [
            'Individuals looking to boost efficiency',
            'People struggling with procrastination',
            'Goal-oriented users tracking progress',
            'Anyone developing better productive habits'
        ],
        'finance': [
            'Individuals gaining control of their finances',
            'People working toward savings goals',
            'Anyone understanding spending patterns',
            'Users making informed financial decisions'
        ]
    }

    return ideal_map.get(subcategory, [
        f'Anyone working with {subcategory} tasks regularly',
        'Notion users streamlining their workflows',
        'Individuals valuing organization and efficiency',
        'People seeking comprehensive management solutions'
    ])

def generate_complexity(price):
    """Generate complexity level based on price."""
    if price == 0:
        return {
            'level': 'beginner',
            'description': 'Easy to set up and use, perfect for Notion beginners',
            'setupTime': '5-10 minutes',
            'learningCurve': 'Gentle - intuitive interface with clear structure'
        }
    elif price < 10:
        return {
            'level': 'beginner-intermediate',
            'description': 'Straightforward setup with some advanced features',
            'setupTime': '15-20 minutes',
            'learningCurve': 'Moderate - some Notion familiarity helpful for advanced features'
        }
    else:
        return {
            'level': 'intermediate',
            'description': 'Comprehensive system with advanced customization',
            'setupTime': '20-30 minutes',
            'learningCurve': 'Moderate - best for users comfortable with Notion databases'
        }

def generate_faqs(name, subcategory):
    """Generate 4 FAQs."""
    return [
        {
            'question': 'Do I need a paid Notion account to use this template?',
            'answer': 'No, this template works perfectly with a free Notion account. Simply duplicate it to your workspace and start using it immediately.'
        },
        {
            'question': 'Can I customize this template to fit my specific needs?',
            'answer': 'Absolutely! This template is fully customizable. You can modify the layout, add or remove fields, change views, and adapt it to match your exact requirements.'
        },
        {
            'question': 'Is this template suitable for beginners?',
            'answer': 'Yes, this template is designed to be user-friendly for all Notion skill levels. It includes clear organization and intuitive structure that makes it easy to understand from day one.'
        },
        {
            'question': f'How do I get started with this {subcategory} template?',
            'answer': 'After purchase, you\'ll receive a link to duplicate the template. Click "Duplicate" in the top-right corner of Notion, and it will be added to your workspace ready to use.'
        }
    ]

# Process each template
for idx, template in enumerate(batch, start=300):
    try:
        product_file = CONTENT_DIR / template['file']

        if not product_file.exists():
            print(f"  ⚠️  [{idx}] File not found: {template['file']}")
            errors.append({'file': template['file'], 'error': 'File not found'})
            continue

        # Read product
        with open(product_file, 'r') as f:
            product = json.load(f)

        # Skip if already enhanced
        if 'enhancedContent' in product:
            print(f"  ⏭️  [{idx}] Skipped (already enhanced): {product.get('name', 'Unknown')}")
            skipped_count += 1
            continue

        # Generate enhanced content
        enhanced_content = generate_enhanced_content(product)

        # Add to product
        product['enhancedContent'] = enhanced_content
        product['lastUpdated'] = datetime.utcnow().isoformat() + 'Z'

        # Write back
        with open(product_file, 'w') as f:
            json.dump(product, f, indent=2, ensure_ascii=False)

        print(f"  ✅ [{idx}] Enhanced: {product.get('name', 'Unknown')}")
        enhanced_count += 1

    except Exception as e:
        print(f"  ❌ [{idx}] Error processing {template.get('file', 'unknown')}: {str(e)}")
        errors.append({'file': template.get('file', 'unknown'), 'error': str(e)})

# Summary
print('\n' + '='*60)
print('BATCH 300-349 PROCESSING COMPLETE')
print('='*60)
print(f'✅ Enhanced: {enhanced_count}')
print(f'⏭️  Skipped: {skipped_count}')
print(f'❌ Errors: {len(errors)}')

if errors:
    print('\nErrors:')
    for err in errors:
        print(f"  - {err['file']}: {err['error']}")

File: scripts/enhance-batch-300-349.js
const fs = require('fs');
const path = require('path');

// Load the templates to enhance
const templatesFile = './templates-to-enhance.json';
const templates = JSON.parse(fs.readFileSync(templatesFile, 'utf8'));

// Extract batch 300-349
const batch = templates.slice(300, 350);

console.log(`Processing batch 300-349: ${batch.length} templates`);

let enhanced = 0;
let skipped = 0;
let errors = [];

function generateEnhancedContent(product) {
  const name = product.name;
  const description = product.description || '';
  const subcategory = product.subcategory || '';
  const price = product.price || 0;

  // Generate overview (2-3 paragraphs)
  const overview = generateOverview(name, description, subcategory, price);

  // Generate feature categories
  const featureCategories = generateFeatureCategories(name, description, subcategory);

  // Generate problems and solutions
  const problemsSolved = generateProblemsSolved(subcategory, name);
  const solutionsBenefits = generateSolutionsBenefits(subcategory, name);

  // Generate ideal for / not recommended
  const idealFor = generateIdealFor(subcategory, name);
  const notRecommendedFor = generateNotRecommendedFor(subcategory, name);

  // Generate complexity level
  const complexityLevel = generateComplexityLevel(subcategory, price);

  // Generate FAQs
  const faqs = generateFAQs(name, subcategory);

  return {
    overview,
    featureCategories,
    problemsSolved,
    solutionsBenefits,
    idealFor,
    notRecommendedFor,
    complexityLevel,
    faqs
  };
}

function generateOverview(name, description, subcategory, price) {
  const cleanDesc = description.substring(0, 500).replace(/\s+/g, ' ').trim();

  return [
    `The ${name} is a powerful Notion ${subcategory} designed to streamline your workflow and boost productivity. ${cleanDesc.includes('template') ? 'This template' : 'This solution'} provides a comprehensive system for managing your ${subcategory}-related tasks with ease and efficiency.`,

    `Built with user experience in mind, this ${subcategory} template offers an intuitive interface that makes it easy to get started immediately. Whether you're a Notion beginner or an experienced user, you'll find the structure logical and the features accessible.`,

    `${price === 0 ? 'Available as a free resource, this template delivers exceptional value without any financial commitment.' : `At $${price}, this template offers excellent value for anyone looking to enhance their ${subcategory} workflow.`} The template includes all the essential features you need to stay organized and achieve your goals.`
  ];
}

function generateFeatureCategories(name, description, subcategory) {
  const categories = [];

  // Core features
  categories.push({
    name: 'Core Features',
    icon: '⚡',
    features: [
      `Comprehensive ${subcategory} management system`,
      'Intuitive database structure with multiple views',
      'Customizable fields and properties',
      'Easy-to-use interface for quick navigation'
    ]
  });

  // Organization
  categories.push({
    name: 'Organization',
    icon: '📊',
    features: [
      'Multiple database views (table, board, calendar)',
      'Advanced filtering and sorting options',
      'Tag and category system',
      'Quick search and navigation'
    ]
  });

  // Productivity
  categories.push({
    name: 'Productivity',
    icon: '🎯',
    features: [
      'Template buttons for quick entry creation',
      'Automated formulas and calculations',
      'Progress tracking and status updates',
      'Integration with other Notion databases'
    ]
  });

  // Customization
  categories.push({
    name: 'Customization',
    icon: '🎨',
    features: [
      'Fully customizable layout and design',
      'Flexible database properties',
      'Aesthetic and functional design',
      'Adaptable to your specific workflow'
    ]
  });

  return categories;
}

function generateProblemsSolved(subcategory, name) {
  const problemMap = {
    'tracker': [
      'Difficulty keeping track of multiple items or activities',
      'Lost or forgotten information',
      'Scattered data across multiple tools',
      'Lack of visibility into patterns and trends'
    ],
    'planner': [
      'Poor time management and scheduling conflicts',
      'Forgotten tasks and missed deadlines',
      'Inability to prioritize effectively',
      'Lack of overview of upcoming commitments'
    ],
    'dashboard': [
      'Information scattered across multiple sources',
      'No centralized view of important data',
      'Time wasted switching between different tools',
      'Difficulty maintaining an overview of projects'
    ],
    'productivity': [
      'Low productivity and wasted time',
      'Unclear priorities and goals',
      'Difficulty maintaining focus',
      'Lack of progress tracking'
    ],
    'finance': [
      'Poor visibility into spending patterns',
      'Difficulty budgeting and saving',
      'Disorganized financial records',
      'Lack of financial planning'
    ],
    'second-brain': [
      'Information overload and mental clutter',
      'Forgotten ideas and knowledge',
      'Difficulty connecting related concepts',
      'Inefficient knowledge management'
    ],
    'default': [
      `Disorganized ${subcategory} management`,
      'Time wasted on manual processes',
      'Lack of systematic approach',
      'Difficulty maintaining consistency'
    ]
  };

  return problemMap[subcategory] || problemMap['default'];
}

function generateSolutionsBenefits(subcategory, name) {
  const benefitMap = {
    'tracker': [
      'Complete visibility of all tracked items in one place',
      'Never lose important information again',
      'Identify patterns and trends easily',
      'Make data-driven decisions with confidence'
    ],
    'planner': [
      'Clear overview of your schedule and commitments',
      'Never miss important deadlines or events',
      'Prioritize tasks effectively',
      'Reduce stress with better planning'
    ],
    'dashboard': [
      'All your important information in one centralized location',
      'Save time by eliminating tool-switching',
      'Quick access to critical data',
      'Better decision-making with complete visibility'
    ],
    'productivity': [
      'Boost your productivity and accomplish more',
      'Clear priorities and focused work',
      'Track progress toward your goals',
      'Develop sustainable productive habits'
    ],
    'finance': [
      'Complete control over your finances',
      'Make informed financial decisions',
      'Achieve your savings goals faster',
      'Reduce financial stress and anxiety'
    ],
    'second-brain': [
      'Capture and organize all your knowledge',
      'Never forget important ideas',
      'Connect concepts and spark creativity',
      'Build a valuable personal knowledge base'
    ],
    'default': [
      `Streamlined ${subcategory} workflow`,
      'Increased efficiency and organization',
      'Better results with less effort',
      'Peace of mind with systematic management'
    ]
  };

  return benefitMap[subcategory] || benefitMap['default'];
}

function generateIdealFor(subcategory, name) {
  const idealMap = {
    'tracker': [
      'Anyone who needs to monitor multiple items or activities',
      'Detail-oriented individuals who value organization',
      'People who want to identify patterns in their data',
      'Users looking for a centralized tracking solution'
    ],
    'planner': [
      'Busy professionals managing multiple commitments',
      'Students balancing coursework and activities',
      'Anyone who wants to improve their time management',
      'People who prefer digital planning solutions'
    ],
    'dashboard': [
      'Professionals who need quick access to key information',
      'Project managers coordinating multiple initiatives',
      'Entrepreneurs managing various business aspects',
      'Anyone who values having a centralized workspace'
    ],
    'productivity': [
      'Individuals looking to boost their efficiency',
      'People struggling with procrastination',
      'Goal-oriented users who want to track progress',
      'Anyone seeking to develop better productive habits'
    ],
    'finance': [
      'Individuals wanting to gain control of their finances',
      'People working toward specific savings goals',
      'Anyone looking to understand their spending patterns',
      'Users who want to make informed financial decisions'
    ],
    'default': [
      `Anyone working with ${subcategory} tasks regularly`,
      'Notion users looking to streamline their workflow',
      'Individuals who value organization and efficiency',
      'People seeking a comprehensive management solution'
    ]
  };

  return idealMap[subcategory] || idealMap['default'];
}

function generateNotRecommendedFor(subcategory, name) {
  return [
    'Users who prefer physical/paper-based systems',
    'People uncomfortable with digital tools',
    'Those looking for a mobile-first solution (Notion mobile has limitations)',
    'Users who need offline access as primary feature'
  ];
}

function generateComplexityLevel(subcategory, price) {
  // Free templates tend to be simpler
  if (price === 0) {
    return {
      level: 'beginner',
      description: 'Easy to set up and use, perfect for Notion beginners',
      setupTime: '5-10 minutes',
      learningCurve: 'Gentle - intuitive interface with clear structure'
    };
  } else if (price < 10) {
    return {
      level: 'beginner-intermediate',
      description: 'Straightforward setup with some advanced features to grow into',
      setupTime: '15-20 minutes',
      learningCurve: 'Moderate - may require some Notion familiarity for advanced features'
    };
  } else {
    return {
      level: 'intermediate',
      description: 'Comprehensive system with advanced features and customization options',
      setupTime: '20-30 minutes',
      learningCurve: 'Moderate - best suited for users comfortable with Notion databases'
    };
  }
}

function generateFAQs(name, subcategory) {
  return [
    {
      question: 'Do I need a paid Notion account to use this template?',
      answer: 'No, this template works perfectly with a free Notion account. Simply duplicate it to your workspace and start using it immediately.'
    },
    {
      question: 'Can I customize this template to fit my specific needs?',
      answer: 'Absolutely! This template is fully customizable. You can modify the layout, add or remove fields, change views, and adapt it to match your exact workflow requirements.'
    },
    {
      question: 'Is this template suitable for beginners?',
      answer: 'Yes, this template is designed to be user-friendly for Notion users of all levels. It includes clear structure and organization that makes it easy to understand and use from day one.'
    },
    {
      question: `How do I get started with this ${subcategory} template?`,
      answer: 'After purchasing, you\'ll receive a link to duplicate the template into your Notion workspace. Click "Duplicate" in the top-right corner, and the template will be added to your workspace ready to use.'
    }
  ];
}

// Process each template
for (const template of batch) {
  try {
    const productPath = path.join(__dirname, '..', 'content', 'products', template.file);

    // Check if file exists
    if (!fs.existsSync(productPath)) {
      console.log(`  ⚠️  File not found: ${template.file}`);
      errors.push({ file: template.file, error: 'File not found' });
      continue;
    }

    // Read the product file
    const product = JSON.parse(fs.readFileSync(productPath, 'utf8'));

    // Check if already has enhancedContent
    if (product.enhancedContent) {
      console.log(`  ⏭️  Skipped (already enhanced): ${product.name}`);
      skipped++;
      continue;
    }

    // Generate enhanced content
    const enhancedContent = generateEnhancedContent(product);

    // Add enhanced content to product
    product.enhancedContent = enhancedContent;
    product.lastUpdated = new Date().toISOString();

    // Write back to file
    fs.writeFileSync(productPath, JSON.stringify(product, null, 2));

    console.log(`  ✅ Enhanced: ${product.name}`);
    enhanced++;

  } catch (error) {
    console.error(`  ❌ Error processing ${template.file}:`, error.message);
    errors.push({ file: template.file, error: error.message });
  }
}

// Summary
console.log('\n' + '='.repeat(60));
console.log('BATCH 300-349 PROCESSING COMPLETE');
console.log('='.repeat(60));
console.log(`✅ Enhanced: ${enhanced}`);
console.log(`⏭️  Skipped: ${skipped}`);
console.log(`❌ Errors: ${errors.length}`);

if (errors.length > 0) {
  console.log('\nErrors:');
  errors.forEach(err => {
    console.log(`  - ${err.file}: ${err.error}`);
  });
}
